File: codes/java/src/main/java/com/github/nicolas_grevin/green_it/model/Astronaut.java
package com.github.nicolas_grevin.green_it.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "astronauts")
final public class Astronaut {
  
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private long id;

  @Column(name = "name", nullable = false, length = 50, unique = true)
  private String name;

  public Astronaut() {

  }

  public Astronaut(long id, String name) {
    super();
    this.id = id;
    this.name = name;
  }

  public long getId() {
    return this.id;
  }

  public void setId(long id) {
    this.id = id;
  }

  public String getName() {
    return this.name;
  }

  public void setName(String name) {
    this.name = name;
  }
}

File: codes/php/src/Entity/Astronaut.php
<?php

declare(strict_types=1);

namespace App\Entity;

use App\Repository\AstronautRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: AstronautRepository::class)]
#[ORM\Table(name: 'astronauts')]
final class Astronaut
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id;

    #[ORM\Column(type: Types::TEXT, length: 50, unique: true)]
    private ?string $name;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getName(): ?string
    {
        return $this->name;
    }

    public function setName(string $name): Astronaut
    {
        $this->name = $name;

        return $this;
    }
}
